api/models/graphite.go

```diff
@@ -0,0 +1,94 @@
+package models
+
+import (
+	"gopkg.in/raintank/schema.v1"
+	"sort"
+)
+
+type GraphiteRender struct {
+	MaxDataPoints uint32   `json:"maxDataPoints" form:"maxDataPoints"`
+	Targets       []string `json:"target" form:"target" binding:"required"`
+	From          string   `json:"from" form:"from"`
+	Until         string   `json:"until" form:"until"`
+	To            string   `json:"to" form:"to"`
+	Format        string   `json:"format" form:"format"`
+}
+
+type GraphiteFind struct {
+	Query  string `json:"query" form:"query" binding:"required"`
+	From   int64  `json:"from" form:"from"`
+	Until  int64  `json:"until" form:"until"`
+	Format string `json:"format" form:"format"`
+	Jsonp  string `json:"jsonp" form:"jsonp"`
+}
+
+type MetricsDelete struct {
+	Query string `json:"query" form:"query" binding:"required"`
+}
+
+type MetricDefs []schema.MetricDefinition
+
+func (defs MetricDefs) MarshalJSONFast(b []byte) ([]byte, error) {
+	seen := make(map[string]struct{})
+
+	names := make([]string, 0, len(defs))
+
+	for i := 0; i < len(defs); i++ {
+		_, ok := seen[defs[i].Name]
+		if !ok {
+			names = append(names, defs[i].Name)
+			seen[defs[i].Name] = struct{}{}
+		}
+	}
+	sort.Strings(names)
+	b = append(b, '[')
+	for _, name := range names {
+		b = append(b, '"')
+		b = append(b, name...)
+		b = append(b, `",`...)
+	}
+	if len(defs) != 0 {
+		b = b[:len(b)-1] // cut last comma
+	}
+	b = append(b, ']')
+	return b, nil
+}
+
+func (defs MetricDefs) MarshalJSON() ([]byte, error) {
```

Comment:
I find this misleading (the type MetricDefs, and the two json marshal methods). this is all geared towards graphite's metrics/index.json endpoint specifically, we're returning a list of uniqued names found amongst the metricdefinitions, which is the format for only index.json.  We're only serializing 1 out of 11 properties of the MetricDefinition, the way this type and the methods are named, it suggests this code is what you need if you want to serialize metricdefinitions.

Not sure what's the solution, maybe renaming the MetricDefs type to MetricNames or something